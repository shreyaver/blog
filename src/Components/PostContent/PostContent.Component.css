/* .Post-content {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: space-evenly;
    width: 100%;
} */
.Image-panel {
    width:100%;
    height:300px;
    background-position: center; 
    background-repeat: no-repeat;
    background-size: cover;
}
.Post-text {
    padding: 10px 10px 0px 10px;
}
.Date-posted {
    float:left;
}
.Reading-time {
    float:right;
}
.Post-text label {
    display: inline-block;
    margin-top: 10px;
    margin-bottom: 10px;
    height: 55px;
    color: black;
    font-size: 23px;
    overflow: hidden;
}
.Post-text p {
    margin-bottom: 0px;
    margin-top: 0px; 
    overflow: hidden;
    height: 55px;
}