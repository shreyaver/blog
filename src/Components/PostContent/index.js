import React from 'react';
import './PostContent.css';

const PostContent = (props) => {
   const imageData = require(`../../Images/${props.cardData.image}`);
   return (
    <div className = "Post-content">
        <div className= "Image-panel" style={{backgroundImage: `url(${imageData})`}}>
        </div>
        <article className="Post-text">
            <div> 
                <span className = "Date-posted">
                    {props.cardData.date}
                </span>
                <span className="Reading-time">            
                    {props.cardData.readingTime}
                </span>
            </div>
            <label>
                {props.cardData.title}
            </label>
            <p>
                {props.cardData.description}
            </p>
        </article>
    </div>
   )
}
export default PostContent;
