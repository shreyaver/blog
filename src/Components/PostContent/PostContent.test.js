import React from 'react';
import renderer from 'react-test-renderer';
import PostContent from '.';

describe('PostContent', () => {
  const cardData = {
    "date": "2nd Januray, 2018",
    "readingTime": "2 mins",
    "title": "The future of abstract art and the culture ...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 10,
    "liked": false,
    "image": "abstract.png"
  }
  it('renders without crashing', () => {
    const postContent = renderer.create(<PostContent cardData = {cardData} />).toJSON();
    expect(postContent).toMatchSnapshot();
  });
})