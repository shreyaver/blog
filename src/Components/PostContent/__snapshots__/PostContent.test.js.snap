// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostContent renders without crashing 1`] = `
<div
  className="Post-content"
>
  <div
    className="Image-panel"
    style={
      Object {
        "backgroundImage": "url(abstract.png)",
      }
    }
  />
  <article
    className="Post-text"
  >
    <div>
      <span
        className="Date-posted"
      >
        2nd Januray, 2018
      </span>
      <span
        className="Reading-time"
      >
        2 mins
      </span>
    </div>
    <label>
      The future of abstract art and the culture ...
    </label>
    <p>
      Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
    </p>
  </article>
</div>
`;
